regression-test/suites/nereids_p0/datatype/test_cast.groovy

```diff
@@ -96,6 +96,124 @@ suite("test_cast") {
         sql "select cast(true as date);"
         result([[null]])
     }
+
+    explain {
+        sql("select cast('12:00:00' as time);")
+        notContains "cast("
+    }
+
+    testFoldConst("select cast(cast('16:32:04' as time) as string);")
+
+    check_fold_consistency "cast(cast('11:11:11' as time) as bigint);"
+    check_fold_consistency "cast(cast('11:11:11' as time) as char);"
+    check_fold_consistency "cast(cast('11:11:11' as time) as double);"
+    check_fold_consistency "cast(cast('11:11:11' as time) as float);"
+    check_fold_consistency "cast(cast('11:11:11' as time) as int);"
+    check_fold_consistency "cast(cast('11:11:11' as time) as integer);"
+    check_fold_consistency "cast(cast('11:11:11' as time) as json);"
```

Comment:
what's the behaviour of casting to json? as an element? for all `check_fold_consistency` we should also add a `qt_sql` command for result consistency.